backend/mttnet/site.go

```diff
@@ -0,0 +1,612 @@
+// Package mttnet exposes the site functions to be exposed over p2p.
+package mttnet
+
+import (
+	"context"
+	"crypto/rand"
+	"encoding/base32"
+	"encoding/json"
+	"fmt"
+	site "mintter/backend/genproto/documents/v1alpha"
+	"mintter/backend/vcs/vcssql"
+	"net/http"
+	"net/url"
+	"reflect"
+	"runtime"
+	"strings"
+	"time"
+
+	"github.com/ipfs/go-cid"
+	"github.com/libp2p/go-libp2p/core/peer"
+	"go.uber.org/zap"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/metadata"
+	rpcpeer "google.golang.org/grpc/peer"
+	emptypb "google.golang.org/protobuf/types/known/emptypb"
+	"google.golang.org/protobuf/types/known/timestamppb"
+)
+
+const (
+	// MttHeader is the headers bearing the remote site hostname to proxy calls to.
+	MttHeader = "x-mintter-site-hostname"
+	// SiteAccountIDCtxKey is the key to pass the account id via context down to a proxied call
+	// In initial site add, the account is not in the database and it needs to proxy to call redeemtoken.
+	SiteAccountIDCtxKey = "x-mintter-site-account-id"
+	// WellKnownPath is the path (to be completed with http(s)+domain) to call to get data from site.
+	WellKnownPath = ".well-known"
```

Comment:
Well known path must have a specifier. The path must be `<hostname>/.well-known/<something-mintter-specific>, can’t be just ending with well-known. It seems like it’s being used in the code without the specified, is it? @juligasa @ericvicenti

Reply:
Exactly, now if you go to https://site.test.mintter.com/.well-known you will find the necessary information to connect. Do you mean it should be https://site.test.mintter.com/.well-known/address or something like that? Or WDYM by specifier @burdiyan ?

Reply:
Shit, something happened to my previous comment, lost the most important part.

The path cannot end at well-known. It has to be `/.well-known/<something>`. Like `/.well-known/mintter` or something like that.

Here's a list of some existing specifiers: https://en.wikipedia.org/wiki/Well-known_URI.

Reply:
Shit, something happened to my previous comment, lost the most important part.

The path cannot end at well-known. It has to be `/.well-known/<something>`. Like `/.well-known/mintter` or something like that.

Here's a list of some existing specifiers: https://en.wikipedia.org/wiki/Well-known_URI.

Reply:
@juligasa @ericvicenti I think this comment might have been lost among many timeline events in this PR. Adding the comment as a reminder.

Reply:
Do you think it's very crappy to use `https://site.test.mintter.com/api/mintter-well-known` or some other path that the nextjs app can provide so we don't need to route traffic? then all the http responses would be served by the nextjs and no need of the reverse proxy if no ssl termination. Tha'd make the deployment simpler. @ericvicenti @burdiyan

Reply:
The most elegant option would be to make it with nextjs middlewares but I have no idea how to implement that. In the meantime, I think we can survive with the not-so-elegant solution :)

Reply:
Maybe, let's serve the well-known path from the Go side, and do the path-based routing in the reverse proxy?